src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/FullScreen/PlatformFullScreenEffect.android.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.ComponentModel;
+using Android.OS;
+using Android.Views;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+using static Xamarin.CommunityToolkit.Effects.FullScreenEffect;
+using static Xamarin.CommunityToolkit.PlatformConfiguration.AndroidSpecific.FullScreenEffect;
+using Effects = Xamarin.CommunityToolkit.Android.Effects;
+
+[assembly: ExportEffect(typeof(Effects.FullScreenEffectRouter), nameof(FullScreenEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.Android.Effects
+{
+	public class FullScreenEffectRouter : PlatformEffect
+	{
+		bool isDetaching;
+
+		protected override void OnAttached()
+		{
+			// should we use WeakEventManager instead?
+			(Element as Page)!.Disappearing += FullScreenEffectRouter_Disappearing;
+
+			// TODO: Remove if not required
+			InitialHasNavigationBar = NavigationPage.GetHasNavigationBar(Element as Page);
+			UpdateStatusBar();
+		}
+
+		void FullScreenEffectRouter_Disappearing(object sender, System.EventArgs e) => ResetStatusBar();
+
+		protected override void OnDetached()
+		{
+			(Element as Page)!.Disappearing -= FullScreenEffectRouter_Disappearing;
+			isDetaching = true;
+			ResetStatusBar();
+		}
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (!args.PropertyName.Equals(ModeProperty.PropertyName) || isDetaching)
```

Comment:
Even after `Detached` is executed, does this method fire?

Reply:
Yes I hit that case that why I added that field, but it only happened when I was in the middle of debugging and set the effect to "Disabled" on a page and try navigating to it initially or after having navigated to a page having IsPersistent on. After restarting the debug session this funny behavior is replaced by the fact that OnModeChanged is not called for that page since the default value is "Disabled".

So I suggest we add another enum value named "Default" just to ensure `OnModeChanged` of pages with effect value set to "Disabled" get triggered, it will be useful when the app is in full screen mode either by a persistent effect on a previous page or by something else, we want that to get triggered so that the effect disable the full screen mode on that page and restore it when disappears.